File: server/models/workday.js
import mongoose from "mongoose"

const workdaySchema=mongoose.Schema({
    user:{
        type:mongoose.Schema.ObjectId,
        ref:"User"
    },
    begin:Date,
    end:Date,
    hoursWorked:{
        type:Number,
        default:0
    },
    report:String,
    typeOfWorkday:{
        type:String,
        enum:["onsite","remote"],
    }

})

const Workday=mongoose.model("workdays",workdaySchema);

export default Workday;

File: client/team_track/lib/models/workday.dart
import 'package:team_track/models/user.dart';

class Workday {
  late User user;
  late DateTime begin;
  late DateTime end;
  late double hoursWorked;
  late String report;
  late String typeOfWorkday;
  late double hoursRest ;

  Workday({
    required this.user,
    required this.begin,
    required this.end,
    required this.hoursWorked,
    required this.report,
    required this.typeOfWorkday,
    required this.hoursRest,
  });
}